src/components/dialogs/parameters/parameters.js

```diff
@@ -106,13 +153,197 @@ export const FluxConventions = {
     IIDM: 'iidm',
     TARGET: 'target',
 };
+
 export const LabelledButton = ({ callback, label, name }) => {
     return (
         <Button onClick={callback} className={name}>
             <FormattedMessage id={label} />
         </Button>
     );
 };
+
+export const useParametersBackend = (
+    user,
+    type,
+    backendFetchProviders,
+    backendFetchProvider,
+    backendFetchDefaultProvider,
+    backendUpdateProvider,
+    backendFetchParameters,
+    backendUpdateParameters
+) => {
+    const studyUuid = useSelector((state) => state.studyUuid);
+
+    const { snackError } = useSnackMessage();
+
+    const [providers, setProviders] = useState([]);
```

Comment:
we can also use useMemo here to store the initial value to avoid unnecessary re-rendering

example:
```
const initialProviders = useMemo(() => [], []);
const initialProvider = useMemo(() => null, []);
const initialParams = useMemo(() => null, []);
const [providers, setProviders] = useState(initialProviders);
const [provider, setProvider] = useState(initialProvider);
const [params, setParams] = useState(initialParams);
```

Reply:
Done for the array. Not sure it is needed for null ?